server/test/test-utils.ts

```diff
@@ -0,0 +1,41 @@
+import { getConnection } from 'typeorm';
+import { CanActivate, ExecutionContext } from '@nestjs/common';
+import { TestingModuleBuilder } from '@nestjs/testing';
+import { AuthUserDto } from '../src/decorators/auth-user.decorator';
+import { JwtAuthGuard } from '../src/modules/immich-jwt/guards/jwt-auth.guard';
+
+type CustomAuthCallback = () => AuthUserDto;
+
+export async function initDb() {
+  return getConnection().synchronize(true);
```

Comment:
Does this action create a database in memory or also on the Postgres container?

I've known some problems that the synchronize feature cannot give the same database structure that we are having using migration. Maybe using migration here instead.

Reply:
I did some research, and unfortunately it is no possible to run Postgres in memory. That's why I added the `docker-compose.test.yml` to run a "test" instance of Postgres for the e2e tests.

What `synchronize` does is to run the DB migrations to create the tables in the DB.

Reply:
Ah right, I saw in one place they where using `syncronize` to create the tables, but that is based in the entities so not actually running migrations 😅 https://medium.com/swlh/migrations-over-synchronize-in-typeorm-2c66bc008e74
Yep I agree, I think it would be safer running migrations with `typeorm migration:run`. I'll do this next (as soon as I get some free time)

Reply:
Thanks, man! no rush here.

Reply:
> I did some research, and unfortunately it is no possible to run Postgres in memory. That's why I added the `docker-compose.test.yml` to run a "test" instance of Postgres for the e2e tests.

OK, no problem here. I can setup a CI/CD flow for testing on a VM and report back to the PR